File: Tests/helpers/XCodeTest/XCodeTestUITests/LogoutTests.swift
import XCTest;
import mParticle_Multiplatform_Tests

class LogoutIosTests: XCTestCase, TestingAwaiter {
    var logoutTests = LogoutTests()
    private var expectations = [String: XCTestExpectation]()

    override func setUpWithError() throws {
        continueAfterFailure = true
    }

    override func setUp() {
        logoutTests.beforeAll(awaiter: self)
    }

    
    //  Start TestingAwaiter implementation
    
    func initializeExpectation(description: String) {
        self.expectations[description] = self.expectation(description: description)
    }
    
    func countdown(description: String) {
        if let expectation = self.expectations[description] {
            expectation.fulfill()
        }
    }
    
    func await(description: String, timeout: Double) {
        if let expectation = self.expectations[description] {
            self.wait(for: [expectation], timeout: timeout)
        }
    }
    
    // End TestingAwaiter implementation
    

    override func tearDownWithError() throws {}

    func testNullRequest() throws {
        try logoutTests.testNullRequest()
    }

    func testKeepExistingMpid() throws {
        try logoutTests.testKeepExistingMpid()
    }

    func testUpdateToNewMpid() throws {
        try logoutTests.testUpdateToNewMpid()
    }

    func testIdentityRequestChangesIds() throws {
        try logoutTests.testIdentityRequestChangesIds()
    }
    
    func testIdentityApiRequestWithNullUser() throws {
        try logoutTests.testIdentityApiRequestWithNullUser()
    }

    func testIdentityApiRequestWithCurrentUser() throws {
        try logoutTests.testIdentityApiRequestWithCurrentUser()
    }

    func testIdentityApiRequestWithCurrentUserWithIds() throws {
        try logoutTests.testIdentityApiRequestWithCurrentUserWithIds()
    }

    func testIdentityRequestWithIdsWithCurrentUserWithIds() throws {
        try logoutTests.testIdentityRequestWithIdsWithCurrentUserWithIds()
    }
    
    func createIdentityRequestBeforeAnotherRequestIsMade() throws {
        try logoutTests.createIdentityRequestBeforeAnotherRequestIsMade()
    }
    
    func testIdentityRequestWithIdsWhileCurrentUserWithIds() throws {
        try logoutTests.testIdentityRequestWithIdsWhileCurrentUserWithIds()
    }
}

File: Tests/CommonTests/identity/LogoutTests.kt
package com.mparticle.identity

import com.mparticle.testing.mockserver.EndpointType
import com.mparticle.testing.BaseStartedTest
import kotlin.test.Test

class LogoutTests(): BaseStartedTest(), IdentityTester {

    val identityTester: IdentityTests = IdentityTests(
        identify_login_logout = { logout(it) },
        identify_login_logoutEndpoint = EndpointType.Identity_Logout
    )

    @Test
    override fun testNullRequest() {
        identityTester.testNullRequest()
    }

    @Test
    override fun testIdentityApiRequestWithNullUser() {
        identityTester.testIdentityApiRequestWithNullUser()
    }

    @Test
    override fun testIdentityApiRequestWithCurrentUser() {
        identityTester.testIdentityApiRequestWithCurrentUser()
    }

    @Test
    override fun testIdentityApiRequestWithCurrentUserWithIds() {
        identityTester.testIdentityApiRequestWithCurrentUserWithIds()
    }

    @Test
    override fun testIdentityRequestWithIdsWhileCurrentUserWithIds() {
        identityTester.testIdentityRequestWithIdsWhileCurrentUserWithIds()
    }

    @Test
    override fun testIdentityRequestChangesIds() {
        identityTester.testIdentityRequestChangesIds()
    }

    @Test
    override fun testIdentityRequestWithIdsWithCurrentUserWithIds() {
        identityTester.testIdentityRequestWithIdsWithCurrentUserWithIds()
    }

    @Test
    override fun createIdentityRequestBeforeAnotherRequestIsMade() {
        identityTester.createIdentityRequestBeforeAnotherRequestIsMade()
    }

    @Test
    override fun testUpdateToNewMpid() {
        identityTester.testUpdateToNewMpid()
    }

    @Test
    override fun testKeepExistingMpid() {
        identityTester.testKeepExistingMpid()
    }
}